pliers/extractors/text_encoding.py

```diff
@@ -0,0 +1,812 @@
+        else:
+            self.binary=binary
+        
+        if not isinstance(corpus,str):
+            raise ValueError('Choice of corpus' + \
+             ' should be string or unused')
+
+        self.stop_words = nltk.corpus.stopwords.words('english')
+
+        if not isinstance(stopWords,list):
+            
+            if stopWords !=None:
+                raise ValueError('stopwords should be a list. Default' + \
+                             ' is NLTK-stopwords. If you don\'t want to use + \
+                                any stopwords use content_only = False instead')
+        else:
+            self.stop_words =  stopWords  
+        
+        if embedding.lower() not in self._available_word_embeddings:
+            raise ValueError('selected embedding type is not ' + \
+                             ' compatible with the embedding method, please check the available' +\
+                             ' embeddings')
+        else:
+            self.embedding = embedding
+    
+        ''' Check whether the embedding file exist '''
+        if embedding == self._fasttext:
+            '''for fasttext we only support the following 
+               settings. Thus any other selection (i.e., 
+               dimensionality < 300 or > 300) will be 
+               superseded '''
+            self.corpus = self._fasttext_corpus
+            dimensionality = 300 
+        
+        elif embedding == self._word2vec:
+            '''for word2vec we only support the following 
+               settings. Thus any other selection (i.e., 
+               dimensionality < 300 or > 300) will be 
+               superseded '''
+            self.corpus = self._word2vec_corpus
+            dimensionality = 300 
+        else:
+            self.corpus = corpus
+        
+        self.embedding_input =  self.embedding + self.corpus  + str(dimensionality) 
+        self.embedding_path = os.path.join(self._embedding_model_path,self.embedding)
+        self.embedding_file  = self.embedding_input + self._text
+        self.embedding_file  = os.path.join(self.embedding_path,self.embedding_file)
+
+        if not os.path.exists(self.embedding_file):
+            raise ValueError('Embedding model file ' + self.embedding_file +  
+                                ' is missing. Please download ' + \
+                                'the model file(s) via running the following command: ' +\
+                                'python download.py [embedding_name], where [embedding_name] ' +\
+                                'is from [glove,word2vec,fasttext]. Also see README for ' + \
+                                'the available models.')
+
+
+    def _loadModel(self):
+        self.wvModel = keyedvectors.KeyedVectors.load_word2vec_format(self.embedding_file,binary=self.binary,encoding='latin1')
+        logger.info(self.embedding + ': ' + 'model loaded')
+
+    def _getModel(self):
+        return self.wvModel 
+
+    def getVector(self,stim,cbow=False):
```

Comment:
It looks like for the most part you've already set each class to have an overlapping set of common methods. This will probably make it easier to refactor as I suggest above.